presto-native-execution/presto_cpp/presto_protocol/topological.py

```diff
@@ -29,21 +29,23 @@
 
 
 def topological(graph):
-    order, enter, state = deque(), set(graph), {}
+    order, enter, state = deque(), sorted(graph), {}
 
     def dfs(node):
         state[node] = GRAY
-        for k in graph.get(node, ()):
+        adjacency_list = sorted(graph.get(node, ()))
+        for k in adjacency_list:
             sk = state.get(k, None)
             if sk == GRAY:
                 raise ValueError(f"cycle: {node} {k}")
             if sk == BLACK:
                 continue
-            enter.discard(k)
+            if k in enter:
```

Comment:
@abhiseksaikia : Don't particularly understand why this is changed, but will accept your call on this.

Reply:
Actually earlier it was a map and we converted to a sorted list of keys from the map, so to avoid errors like ValueError, this was needed here.